src/main/java/io/beekeeper/pingdom/resources/PingdomHealthResource.java

```diff
@@ -101,6 +104,9 @@ public PingdomHealth get(@QueryParam("key") String key, @QueryParam("severity")
                                 .append("Reason: ").append(result.getMessage()).append('\n')
                                 .append("Exception: ").append(result.getError() != null ? result.getError().getMessage() : null)
                                 .append('\n');
+                        if (log.isWarnEnabled()) {
+                            log.warn(status.toString());
```

Comment:
this is incorrect. Please see that we are in a loop. We have to do one of 2 things:

1) Print every single status separately. Here, status is a combined object of all previous statuses.
2) Print only at the end if the status combined string is not empty.

otherwise the log will look like:

WARN: Failure 1
WARN: Failure 1... Failure 2
WARN: Failure 1 ... Failure 2 ... Failiure 3

Reply:
Fixed by b4acf57. I implemented the option 2 because it is probably more convenient to see the whole error reported to pingdom as a single log message.

I also found a test where you can see that logs were printed multiple times: testResourceReturnsNotOKWithSeverityAndMultipleChecks2. Now it shows correct output.